src/Files.App/Shell/ShellLibraryEx.cs

```diff
@@ -0,0 +1,303 @@
+		/// Initializes a new instance of the <see cref="ShellLibrary"/> class.
+		/// </summary>
+		/// <param name="iItem">The i item.</param>
+		/// <param name="readOnly">if set to <c>true</c> [read only].</param>
+		public ShellLibraryEx(Shell32.IShellItem iItem, bool readOnly = false)
+		{
+			_lib = new Shell32.IShellLibrary();
+			_lib.LoadLibraryFromItem(iItem, readOnly ? STGM.STGM_READ : STGM.STGM_READWRITE);
+
+			Init(iItem);
+		}
+
+		/// <summary>
+		/// Gets or sets the default target folder the library uses for save operations.
+		/// </summary>
+		/// <value>The default save folder.</value>
+		public ShellItem DefaultSaveFolder
+		{
+			get => Open(_lib.GetDefaultSaveFolder<Shell32.IShellItem>(Shell32.DEFAULTSAVEFOLDERTYPE.DSFT_DETECT));
+			set => _lib.SetDefaultSaveFolder(Shell32.DEFAULTSAVEFOLDERTYPE.DSFT_DETECT, value.IShellItem);
+		}
+
+		/// <summary>Gets the set of child folders that are contained in the library.</summary>
+		/// <value>A <see cref="ShellItemArray"/> containing the child folders.</value>
+		public ShellLibraryFolders Folders
+			=> _folders ??= GetFilteredFolders();
+
+		/// <summary>
+		/// Gets or sets a string that describes the location of the default icon.
+		/// The string must be formatted as
+		/// <c>ModuleFileName,ResourceIndex or ModuleFileName,-ResourceID</c>.
+		/// </summary>
+		/// <value>
+		/// The default icon location.
+		/// </value>
+		public IconLocation IconLocation
+		{
+			get
+			{
+				_ = IconLocation.TryParse(_lib.GetIcon(), out var l);
+				return l;
+			}
+			set => _lib.SetIcon(value.ToString());
+		}
+
+		/// <summary>
+		/// Gets the name relative to the parent for the item.
+		/// </summary>
+		public override string Name { get => _name; protected set => _name = value; }
+
+		/// <summary>
+		/// Gets or sets a value indicating whether to pin the library to the navigation pane.
+		/// </summary>
+		/// <value>
+		/// <c>true</c> if pinned to the navigation pane; otherwise, <c>false</c>.
+		/// </value>
+		public bool PinnedToNavigationPane
+		{
+			get => _lib.GetOptions().IsFlagSet(Shell32.LIBRARYOPTIONFLAGS.LOF_PINNEDTONAVPANE);
+			set => _lib.SetOptions(Shell32.LIBRARYOPTIONFLAGS.LOF_PINNEDTONAVPANE, value ? Shell32.LIBRARYOPTIONFLAGS.LOF_PINNEDTONAVPANE : 0);
+		}
+
+		/// <summary>
+		/// Gets or sets the library's View Template.
+		/// </summary>
+		/// <value>
+		/// The View Template.
+		/// </value>
+		public LibraryViewTemplate ViewTemplate
+		{
+			get => (LibraryViewTemplate)ShlGuidExt.Lookup<Shell32.FOLDERTYPEID>(ViewTemplateId);
+			set
+			{
+				if (value != LibraryViewTemplate.Custom)
+					ViewTemplateId = ((Shell32.FOLDERTYPEID)value).Guid();
+			}
+		}
+
+		/// <summary>
+		/// Gets or sets the library's View Template identifier.
+		/// </summary>
+		/// <value>
+		/// The View Template identifier.
+		/// </value>
+		public Guid ViewTemplateId
+		{
+			get => _lib.GetFolderType();
+			set => _lib.SetFolderType(value);
+		}
+
+		public void Reload()
+		{
```

Comment:
This method is the only one not documented

Reply:
Fixed